src/main/src/main/java/org/geoserver/catalog/impl/DefaultCatalogFacade.java

```diff
@@ -1035,7 +1035,7 @@ public void syncTo(CatalogFacade dao) {
         if (dao instanceof DefaultCatalogFacade) {
             // do an optimized sync
             DefaultCatalogFacade other = (DefaultCatalogFacade) dao;
-
+            Catalog catalog = other.getCatalog();
```

Comment:
Note this fixes a pre-existing issue by which all loaded `StoreInfo`s and `ResourceInfo`s end up with a reference to the discarded `Catalog`, which is then never garbage collected. 
Everything kept working as expected by accident, due to the fact that `CatalogImpl.sync()` will take ownership of the discarded catalog internals (resource pool, facade's CatalogInfoLookup's, etc)